### PR TITLE
fix: Improve the input's width to accommodate full URL

### LINKED ISSUE #1784: Bug: urls is cut off when full url is added to insight page
### Describe the bug

Screen Shot 2023-10-02 at 3 10 20 PM

### Steps to reproduce

1. View insight page
2. https://github.com/open-sauced/app
3. not it is cut off

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by RitaDee**
## Description
This PR adds a new prop `style` to the `Search` component in `search.tsx` and sets the `width` style property to `100%` in the `InsightPage` component in `InsightPage.tsx`. This allows for custom styling of the `Search` component and ensures that the `InsightPage` component takes up the full width.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1784

## Mobile & Desktop Screenshots/Recordings

Before:

https://github.com/open-sauced/app/assets/106407013/14b50119-6de4-406e-b3e7-ca5c09dcd9a6

After:

https://github.com/open-sauced/app/assets/106407013/53effe95-540f-474c-9d4d-f2181e62ad15

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/organisms/InsightPage/InsightPage.tsx [hunk 1]
```diff
@@ -460,6 +460,7 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
             isLoading={createLoading}
             placeholder="Repository URL or Full Name (ex: open-sauced/open-sauced)"
             className="!w-full text-md text-gra"
+            style={{ width: "100%" }}
```
Comment (nickytonline) [inline components/organisms/InsightPage/InsightPage.tsx:463]: Refactor (blocking): The style is not required. The `w-full` Tailwind class does the same thing. It's already been used with a higher CSS specificity via `!w-full`.

It appears to not be applied for some reason, so I'd investigate that.

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: components/organisms/InsightPage/InsightPage.tsx (hunk 1 above)
Comment (RitaDee) [reply components/organisms/InsightPage/InsightPage.tsx:463]: > Refactor (blocking): The style is not required. The `w-full` Tailwind class does the same thing. It's already been used with a higher CSS specificity via `!w-full`.
> 
> It appears to not be applied for some reason, so I'd investigate that.

Yes, the !w-full isn’t working. So, I came up with the style prop.

Comment (nickytonline) [reply components/organisms/InsightPage/InsightPage.tsx:463]: @RitaDee, what I meant is the `style` fix is masking the real issue, which is `!w-full` is not being applied, so I'd dig into that.

Comment (takanome-dev) [reply components/organisms/InsightPage/InsightPage.tsx:463]: @RitaDee the `!w-full` is applied to the div containing the input. The input itself has a width of `w-max` that's why it's not expanding to the full width. You change the `w-max` to `w-full` to fix the issue. You may also want to check all the places that component is used to make sure any other issue is caused

Comment (bdougie) [reply components/organisms/InsightPage/InsightPage.tsx:463]: > The input itself has a width of `w-max` that's why it's not expanding to the full width. You change the `w-max` to `w-full` to fix the issue. You may also want to check all the places that component is used to make sure any other issue is caused

Good call here @takanome-dev. We actually caught this in the discord call this morning with @nickytonline

Comment (RitaDee) [reply components/organisms/InsightPage/InsightPage.tsx]: > Refactor (blocking): The style is not required. The `w-full` Tailwind class does the same thing. It's already been used with a higher CSS specificity via `!w-full`.
> 
> It appears to not be applied for some reason, so I'd investigate that.

@nickytonline, thank you for the walkthrough earlier. Having understood that the project should not go beyond the tailwind class, I opted to replace w-max with w-full in the Search component, and I have checked other places where the Search component is used or called, but I can’t find any regression or weird display. Can you check this out in my current commit?

Thank you @takanome-dev.

**Review by jpmcb — APPROVED** [review]
Screenshot 2023-10-03 at 3 29 54 PM

Functionally looks good to me!!! I'll leave it up to the others for code comment

File: components/atoms/Search/search.tsx [hunk 2]
```diff
@@ -32,6 +32,7 @@ const Search = ({
   suggestions,
   onChange,
   isLoading,
+  ...props
```
Comment (takanome-dev) [inline components/atoms/Search/search.tsx:35]:
```suggestion
```
I think all the props have been already destructured here :eyes:

**Review by takanome-dev — COMMENTED** [review]
Just some minor cleanup and this should be good to go

File: components/atoms/Search/search.tsx [hunk 3]
```diff
@@ -69,14 +70,15 @@ const Search = ({
...
-        className="w-max pl-2 placeholder:text-sm focus:outline-none placeholder:text-slate-400"
+        className="w-full pl-2 placeholder:text-sm focus:outline-none placeholder:text-slate-400"
         autoFocus={autoFocus}
         placeholder={placeholder}
         name={name}
         value={search}
         type="search"
         id={name}
         onChange={handleChange}
+        {...props}
```
[the remark this answers is not in the document]
Comment (RitaDee) [reply components/atoms/Search/search.tsx]: @takanome-dev, done. Thanks for the review.

**Review by takanome-dev — APPROVED** [review]

**Review by nickytonline — APPROVED** [review]
Thanks @RitaDee! 🚢

**Pull request merged**